sov-modules/sov-modules-impl/integration-tests/tests/demo/mod.rs

```diff
@@ -0,0 +1,235 @@
+    type Context = C;
+
+    fn dispatch(self, storage: C::Storage, context: &C) -> Result<CallResponse, Error> {
+        match self {
+            RuntimeCall::Election(message) => {
+                let mut election = Election::<C>::new(storage);
+                election.call(message, context)
+            }
+            RuntimeCall::ValueAdder(message) => {
+                let mut value_adder = ValueAdderModule::<C>::new(storage);
+                value_adder.call(message, context)
+            }
+        }
+    }
+}
+
+/// Methods from this trait should be called only once during the rollup deployment.
+trait Genesis {
+    type Context: Context;
+
+    /// Initializes the state of the rollup.
+    // TDOD: genesis should take initial configuration as an argument.
+    fn genesis() -> Result<<<Self as Genesis>::Context as Context>::Storage, Error>;
+}
+
+// This code will be generated by a macro
+impl<C: Context> Genesis for Runtime<C> {
+    type Context = C;
+
+    fn genesis() -> Result<C::Storage, Error> {
+        let storage = C::Storage::default();
+
+        let mut election = Election::<C>::new(storage.clone());
+        election.genesis()?;
+
+        let mut value_adder = ValueAdderModule::<C>::new(storage.clone());
+        value_adder.genesis()?;
+
+        Ok(storage)
+    }
+}
+
+fn decode_dispatchable<C: Context>(
+    data: Vec<u8>,
+) -> Result<impl DispatchCall<Context = C>, anyhow::Error> {
+    let mut data = Cursor::new(data);
+    Ok(RuntimeCall::<C>::decode(&mut data)?)
+}
+
+// This code will be generated by a macro
+// Possible clients:
+// - test client
+// - rest api client
+// - wasm bindings
+// - json abi
+#[derive(Default)]
+struct Client<C: Context> {
+    _phantom: PhantomData<C>,
+}
+
+// This code will be generated by a macro
+impl<C: Context> Client<C> {
+    fn new() -> Self {
+        Self {
+            _phantom: PhantomData,
+        }
+    }
+
+    fn send_election_message(&self, data: <Election<C> as Module>::CallMessage) -> Vec<u8> {
+        let call = RuntimeCall::<C>::Election(data);
+        let mut data = Vec::default();
+        call.encode(&mut data);
+
+        data
+    }
+
+    fn send_value_adder_message(
+        &self,
+        data: <ValueAdderModule<C> as Module>::CallMessage,
+    ) -> Vec<u8> {
+        let call = RuntimeCall::<C>::ValueAdder(data);
+        let mut data = Vec::default();
+        call.encode(&mut data);
+
+        data
+    }
+
+    fn query_election(&self, data: <Election<C> as Module>::QueryMessage) -> Vec<u8> {
+        let query = RuntimeQuery::<C>::Election(data);
+        let mut data = Vec::default();
+        query.encode(&mut data);
+
+        data
+    }
+
+    fn query_value_adder(&self, data: <ValueAdderModule<C> as Module>::QueryMessage) -> Vec<u8> {
+        let query = RuntimeQuery::<C>::ValueAdder(data);
+        let mut data = Vec::default();
+        query.encode(&mut data);
+
+        data
+    }
+}
```

Comment:
nit: all of these functions can use the `encode_to_vec` method from our `Encode` trait

Reply:
done